src/ZODB/Connection.py

```diff
@@ -1073,7 +1073,8 @@ def _resetCache(self):
     def _release_resources(self):
         for c in self.connections.itervalues():
             if c._mvcc_storage:
-                c._storage.release()
+                if c._storage is not None:
+                    c._storage.release()
```

Comment:
When I wrote the fix, I compared a stock Plone page with our page with Relstorage.
The reason why this causes no exception in a stock Plone/Zope page is that the storage used does not provide mvcc_capability, this is why c._mvcc_storage is False. A stock FileStorage depends on the connection to provide MVCC (From: https://github.com/zopefoundation/ZODB/blob/master/src/ZODB/interfaces.py#L1022)

Since Relstorage supports MVCC at storage level, this code finally get executed and fails with multiple, nested databases. Which you already have with a stock Zope Site, because of the tempstorage.